libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -105,9 +105,9 @@ void AP_Motors::set_radio_passthrough(float roll_input, float pitch_input, float
 void AP_Motors::rc_write(uint8_t chan, uint16_t pwm)
 {
     SRV_Channel::Aux_servo_function_t function = SRV_Channels::get_motor_function(chan);
-    if ((1U<<chan) & _motor_pwm_range_mask) {
+    if ((1U<<chan) & _motor_pwm_scaled.mask) {
         // note that PWM_MIN/MAX has been forced to 1000/2000
-        SRV_Channels::set_output_scaled(function, pwm - 1000);
+        SRV_Channels::set_output_scaled(function, pwm - _motor_pwm_scaled.offset);
```

Comment:
this is subtracting two unsigned numbers, pwm could be 1000, offset will be 1500, so will wrap, then it gets cast to float. I'm not at all confident this will work, and may work differently in SITL to STM32

Reply:
`_motor_pwm_scaled.offset` changed to float and `pwm` now cast to float.